contracts/client/Offchain.sol

```diff
@@ -0,0 +1,89 @@
+pragma solidity ^0.5.15;
+
+/**
+    @notice types defined here are passing between users and the client, but not on chain.
+ */
+library Offchain {
+    struct Transfer {
+        uint256 txType;
+        uint256 fromIndex;
+        uint256 toIndex;
+        uint256 amount;
+        uint256 fee;
+        uint256 nonce;
+    }
+
+    function decodeTransfer(bytes memory txBytes)
```

Comment:
Should check if the amount and fee passed here are compressible. Will think about it more in the next pr